BoincRpc/RpcClient.cs

```diff
@@ -758,6 +1039,29 @@ public async Task SetCoreClientConfigAsync(XElement coreClientConfig)
             CheckResponse(await PerformRpcAsync(request));
         }
 
+        /// <summary>
+        /// Report the status of an android device to the client including:
+        /// - whether the device is plugged in via USB or AC power
+        /// - the battery charge level, state, and temperature
+        /// - whether the device is connected via WiFi
+        /// - whether the user is active(e.g.whether the screen is on)
+        /// - the device name(supplied by the user)
+        /// This RPC also serves as a "keep-alive": if the BOINC client doesn't get one in 30 seconds, it will suspend computation.
+        /// So Android GUIs should call it fairly often (every 10 sec at least).
+        /// There is no C++ binding for this RPC; there's a Java binding in the Android GUI code: https://github.com/BOINC/boinc/blob/master/android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/RpcClient.java
+        /// </summary>
+        /// <param name="coreClientConfig"></param>
+        /// <returns></returns>
+        public async Task ReportAndroidDeviceStatusAsync()
```

Comment:
I deliberately left out the implementation of this RPC call as it is specific to Android and even the official C++ bindings do not support it. People writing Android apps would most likely use the existing Java library instead of this one, so I would not include the operation here.

Reply:
Done.